contracts/LockDealNFT/LockDealNFTModifiers.sol

```diff
@@ -36,11 +31,41 @@ abstract contract LockDealNFTModifiers is LockDealState, Ownable {
         _;
     }
 
+    modifier approvedAmount(address token, address from, uint256 amount) {
+        _approvedAmount(token, from, amount);
+        _;
+    }
+
     function _notZeroAddress(address _address) private pure {
         require(_address != address(0x0), "Zero Address is not allowed");
     }
 
+    function _onlyContract(address contractAddress) private view {
```

Comment:
why do we need this ?

Reply:
providers are only contracts

Reply:
when we use internal functions inside modifiers less gas is used